src/NuGet.Protocol.VisualStudio/PSSearchResourceV3.cs

```diff
@@ -1,35 +1,100 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+using NuGet.Packaging.Core;
 using NuGet.Protocol.Core.Types;
+using NuGet.Protocol.Core.v3;
+using NuGet.Versioning;
 
 namespace NuGet.Protocol.VisualStudio
 {
-    public class PSSearchResourceV3 : PSSearchResource
+    public class PowerShellSearchResourceV3 : PSSearchResource
```

Comment:
Should you rename this?! Because, it does not match with the provider now. Also, 'Powershell' prefix has been used by the resources for powershell-get. I believe it gets confusing